app/main.py

```diff
@@ -1,3 +1,72 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> "Distance":
+        if isinstance(other, Distance):
```

Comment:
For better optimization you can try to create a separate method for your code duplicates in all function for checking